implementations/rust/ockam/ockam_vault_sync_core/src/vault_mutex/hasher.rs

```diff
@@ -4,7 +4,17 @@ use ockam_vault_core::{Hasher, Secret, SecretAttributes, SmallBuffer};
 
 impl<V: Hasher> Hasher for VaultMutex<V> {
     fn sha256(&mut self, data: &[u8]) -> Result<[u8; 32]> {
-        self.0.lock().unwrap().sha256(data)
+        #[cfg(feature = "std")]
```

Comment:
That looks very copy&pasty. If we can't make it better, how about disabling VaultMutex for no_std? I think we only use it for testing purposes when we don't want to a start node.

Reply:
Can we please keep those for now, t

Reply:
Can we please keep those for now? I'm busy with a `no_std` Mutex implementation with the same syntax as `std::sync::Mutex` so those are a placeholder until we can move all Mutex usages to `compat::sync::Mutex`

Reply:
Sure